modules/services/barrier.nix

```diff
@@ -0,0 +1,72 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let cfg = config.services.barrier;
+in {
+
+  meta.maintainers = with maintainers; [ kritnich ];
+
+  options.services.barrier = {
+
+    client = {
+
+      enable = mkEnableOption "Barrier Client daemon";
+
+      name = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = ''
+          Screen name of client. Defaults to hostname.
+        '';
+      };
+
+      server = mkOption {
+        type = types.str;
+        default = null;
```

Comment:
The default value of null doesn't work with `types.str`?
Seems unintentionnal

Reply:
You can simply remove the `default = null;` line. That will force the user to supply the value.

Reply:
Just don't specify a default if it's required. Home manager will handle it and print an error.